iree/hal/vmla/op_kernels_generic.h

```diff
@@ -511,6 +514,29 @@ Status Reverse::Execute(absl::Span<const T> src_buffer,
   return OkStatus();
 }
 
+template <typename T>
+Status Sort::Execute(absl::Span<const T> src_buffer,
+                     absl::Span<int32_t> dst_buffer, ShapeSpan src_shape) {
+  int elements = 1;
+  for (auto dim : src_shape) {
+    elements *= dim;
+  }
+
+  const int sort_size = src_shape.back();
+
+  for (int i = 0; i < elements; i += sort_size) {
+    auto src_subspan = src_buffer.subspan(i, sort_size);
+    auto dst_subspan = dst_buffer.subspan(i, sort_size);
+    std::iota(dst_subspan.begin(), dst_subspan.end(), 0);
+    std::stable_sort(dst_subspan.begin(), dst_subspan.end(),
```

Comment:
Do we have plans for how to lower sort to other HAL backends? Seems like something worth having specialized handling for (e.g. https://developer.nvidia.com/gpugems/gpugems2/part-vi-simulation-and-numerical-algorithms/chapter-46-improved-gpu-sorting, https://arxiv.org/pdf/1709.02520.pdf) in a GPU or parallel computing dialect, which would be in line with this lowering strategy (rather than relying on the higher levels to translate `tf.sort` into algorithmic ops).

I worry a bit seeing anything from `<algorithm>` used here in VMLA since there's some pretty complex logic baked into the algorithms implemented there, that generally don't have readability available implementations across our less generic targets.

Reply:
I expect other HAL backends will need to do a completely separate path for two reasons.

1 - Codegen can happily inline any comparison operators, allowing them to directly use the comparison operation defined. VMLA can't as easily do this as the comparison region could be arbitrarily complicated.

2 - The specifics of how sorting is done is completely dependent on the device architecture. E.g. the radix sort can special case some comparisons but would not be useful for floating point. HLO's sort depends on object comparisons which would require an alternative sorting method.

Reply:
SGTM